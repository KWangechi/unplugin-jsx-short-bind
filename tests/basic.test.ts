import path from 'node:path'
import { describe } from 'vitest'
import {
  RollupEsbuildPlugin,
  RollupRemoveVueFilePathPlugin,
  RollupVue,
  rollupBuild,
  testFixtures,
} from '@vue-macros/test-utils'
import JsxShortBind from '../src/rollup'

describe('fixtures', async () => {
  await testFixtures(
    ['tests/fixtures/*.vue'],
    (args, id) => rollupBuild(id, [
      JsxShortBind(),
      RollupVue(),
      RollupRemoveVueFilePathPlugin(),
      RollupEsbuildPlugin({
        target: 'esnext',
      }),
    ]),
    { cwd: path.resolve(__dirname, '..'), promise: true },
  )
})
