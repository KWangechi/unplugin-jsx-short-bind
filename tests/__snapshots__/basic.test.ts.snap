// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fixtures > tests/fixtures/basic.tsx 1`] = `
"// basic.js
import { createVNode, Fragment } from 'vue';

var basic = () => {
  let value = $ref('foo');
  function onInput(e) {
    value = e.target?.value;
  }
  return createVNode(Fragment, null, [createVNode("input", {
    "onInput": onInput,
    "value": value
  }, null)]);
};

export { basic as default };
"
`;
